pkg/plugin/conntrack/_cprog/conntrack.c

```diff
@@ -84,17 +114,16 @@ static __always_inline __u8 _ct_get_traffic_direction(__u8 observation_point) {
  * @arg key The key to be used to create the new connection.
  * @arg flags The flags of the packet.
  * @arg observation_point The point in the network stack where the packet is observed.
- * @arg timeout The timeout for the connection.
  */
-static __always_inline bool _ct_create_new_tcp_connection(struct ct_v4_key key, __u8 flags, __u8 observation_point, __u64 timeout) {
+static __always_inline bool _ct_create_new_tcp_connection(struct ct_v4_key key, __u8 flags, __u8 observation_point) {
     struct ct_entry new_value;
     __builtin_memset(&new_value, 0, sizeof(struct ct_entry));
     __u64 now = bpf_mono_now();
     // Check for overflow
-    if (timeout > UINT64_MAX - now) {
+    if (CT_SYN_TIMEOUT > UINT32_MAX - now) {
```

Comment:
@rectified95 this should still be `UINT64_MAX` right? since `now` is a _u64, although we are going to downcast it to _u32 when setting it to `eviction_time` anyway

Reply:
Looking at Cilium's use of `bpf_mono_now` they're inconsistently using 32 and 64-bit variables.
https://github.com/search?q=repo%3Acilium%2Fcilium%20bpf_mono_now&type=code

Is there a specific reason you're using this function? It looks like it returns its value in 'jiffies' which are linux units related to cpu ticks 1-10ms long?

Reply:
Actually, its definitiona depends on whether  `jiffies`is defined, it also uses bpf_ktime_get_sec conditionally
https://github.com/cilium/cilium/blob/510b222f50256c1e876409387c4b93087586256b/bpf/lib/time.h#L14

Reply:
why don't you just use that instead directly? it returns seconds and you can stick to __u32

Reply:
i think it only returns value in jiffies if `defined(HAVE_JIFFIES) && defined(ENABLE_JIFFIES) && KERNEL_HZ != 1` right? This is that function definition in Retina now
```
#define NSEC_PER_SEC	(1000ULL * 1000ULL * 1000UL)
#define bpf_ktime_get_sec()	\
	({ __u64 __x = bpf_ktime_get_boot_ns() / NSEC_PER_SEC; __x; })
# define bpf_mono_now()		bpf_ktime_get_sec()
```

Reply:
it is in `conntrack.h`

Reply:
To your original question - you should be checking for overflow using the bounds of the final storage type of eviction_tyime, meaning u32